fdbserver/BlobWorker.actor.cpp

```diff
@@ -1279,9 +1279,9 @@ ACTOR Future<BlobFileIndex> compactFromBlob(Reference<BlobWorkerData> bwData,
 		ASSERT(snapshotVersion < version);
 
 		state Optional<BlobGranuleCipherKeysCtx> snapCipherKeysCtx;
-		if (snapshotF.cipherKeysMeta.present()) {
-			ASSERT(bwData->isEncryptionEnabled);
-
+		if (isEncryptionOpSupported(EncryptOperationType::BLOB_GRANULE_ENCRYPTION)) {
```

Comment:
I like the change, but the ordering of the original one was better I guess. It means the readSnapshot is `encrypted` and we assert that cluster configurations also support encryption. In the modified one, it may happen that cluster is mis-configured but we read an encrypted snapshot file. 

I think the change should be: `bwData->isEncryptionEnabled` -> `isEncryptionOpSupported(EncryptOperationType::BLOB_GRANULE_ENCRYPTION)`

Reply:
Sure i can revert it but i think we should have my check during simulation. It will help to potentially catch bugs where we aren't encrypting when we should be